testing/testing/src/main/java/io/helidon/testing/TestConfig.java

```diff
@@ -0,0 +1,243 @@
+     *
+     * @param key   key to set
+     * @param value value to set
+     */
+    public static void set(String key, String value) {
+        LOCK.lock();
+        try {
+            if (io.helidon.common.GlobalInstances.current(Sources.class).isPresent()) {
+                io.helidon.common.GlobalInstances.get(Sources.class, Sources::new)
+                        .sources()
+                        .forEach(it -> it.set(key, value));
+            } else {
+                io.helidon.common.GlobalInstances.get(AccumulatedOptions.class, AccumulatedOptions::new)
+                        .options().put(key, value);
+            }
+        } finally {
+            LOCK.unlock();
+        }
+    }
+
+    static void register(TestConfigSource source) {
+        LOCK.lock();
+        try {
+            if (io.helidon.common.GlobalInstances.current(Sources.class).isEmpty()) {
+                AccumulatedOptions accumulatedOptions = io.helidon.common.GlobalInstances.get(AccumulatedOptions.class,
+                                                                                              AccumulatedOptions::new);
+                accumulatedOptions.options().forEach(source::set);
+                accumulatedOptions.options().clear();
+            }
+            io.helidon.common.GlobalInstances.get(Sources.class, Sources::new)
+                    .sources()
+                    .add(source);
+        } finally {
+            LOCK.unlock();
+        }
+    }
+
+    /**
+     * Config profile to use.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    public @interface Profile {
+        /**
+         * Config profile to set. This will always override profile set by other means.
+         *
+         * @return config profile to use
+         */
+        String value();
+    }
+
+    /**
+     * Custom configuration key/value pair.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    @Repeatable(ConfigValues.class)
+    public @interface Value {
+        /**
+         * Key of the added configuration option.
+         *
+         * @return key
+         */
+        String key();
+
+        /**
+         * Value of the added configuration option.
+         *
+         * @return value
+         */
+        String value();
+
+        /**
+         * Weight of this configuration option.
+         *
+         * @return weight
+         */
+        double weight() default 954_000;
+    }
+
+    /**
+     * Repeatable annotation for {@link io.helidon.testing.TestConfig.Value}.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    public @interface ConfigValues {
+        /**
+         * Config values.
+         *
+         * @return values
+         */
+        Value[] value();
+    }
+
+    /**
+     * Custom configuration key/value pairs.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    public @interface Values {
```

Comment:
How about `Block` to follow the current pattern used in MP testing?

Reply:
Fixed. Also fixed the container annotations to be just plurals of the repeated annotations (so there is still `Values`, but now it is a container for `Value`, and there is a `Block` annotation for config source text block